File: programmers/level-2/sort/H-Index.js
// h번 이상 인용된 논문이 h편 이상, 나머지는 h편 이하 인용
// h번 이상 인용된 논문 수 <= 인용횟수이면 loop -> HIndex++
function solution(citations) {
  let hIndex = 0;
  citations.sort((a, b) => b - a);

  while (hIndex + 1 <= citations[hIndex]) {
    hIndex++;
  }

  return hIndex;
}

solution([3, 0, 6, 1, 5]);

File: programmers/level-2/sort/H-Index.py
# sol1
def solution(citations):
    h = 0
    
    for i in range(max(citations)+1):
        over_count = sum(1 for x in citations if x >= i)
        if over_count >= i:
            h = i
            
    return h

# sol1
def solution(citations):
    citations.sort(reverse=True)
    h = 0
    
    for i, citation in enumerate(citations):
        if citation >= i+1:
            h = i+1
        else:
            break
            
    return h